### PR TITLE
Random

### PR DESCRIPTION
**Pull request by garrettwrong**
Stashing some random cleanup.

This sort of hits a snag. I was building up to moving from leagcy RandomState to Generator, but much of our code is sensitive to sequence of calls in `uniform_random_angles` and I haven't decided how I want to deal with that.

I think it would be reasonable for future development to use `aspire.utils.random` for random needs, and let it handle contexts etc.  In unit tests and user code they can do whatever they want.  Currently we are sometimes having the developer manage context, and sometimes wrapping for them.

I might break this issue up into things we can do now, and things that might make sense to do during or after unit test reformation.

### CODE REVIEW

File: src/aspire/utils/coor_trans.py [hunk 1]
```diff
@@ -153,6 +153,17 @@ def uniform_random_angles(n, seed=None, dtype=np.float32):
             np.arccos(2 * np.random.random(n) - 1),
             np.random.random(n) * 2 * np.pi
         ))
+
+    # TODO, If/when unit tests refactoring allows, we should swap out
+    #   for internal random calls. This will facility moving from
+    #   RandomState to Generator.
+    # angles = np.column_stack((
```
Comment (janden) [inline src/aspire/utils/coor_trans.py]: It's not clear to me what this code is doing here. Ideally, we don't want to be calling `random` several times over with the same seed (i.e., what we're doing above seems more appropriate). Perhaps I'm missing something here.

File: src/aspire/utils/coor_trans.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from scipy.linalg import svd
 
 from aspire.utils import ensure
-from aspire.utils.matlab_compat import Random, randn
+from aspire.utils.random import Random, randn, random
```
Comment (janden) [inline src/aspire/utils/coor_trans.py]: `random` unused.

**Review by janden — CHANGES_REQUESTED** [review]
I'm not 100% I understand what this PR is supposed to do. We're moving the random functions into their own submodule, great. But what's the advantage of going from `np.random.choice` to our own `choice`, for example? Is this with some sort of future changes in mind?

**Comment by garrettwrong** [comment]
>  But what's the advantage of going from `np.random.choice` to our own `choice`, for example? 

I think it is preferable for our module to manage contexts for us.  Otherwise we have `with Random` sprinkled all around the code by developers in various ways, at various levels, and in various sequences (that are currently sensitive, see below)... It is really confusing to follow actually (well for me at least)...  Our current Random context thing seems to do the right thing by external codes (protects their state if we're using a `Random` context), but a less than ideal thing for ASPIRE. It is a great start since we're in the right places already....  I think if we use `Random` consistently it will make managing the randomized stuff more efficient, and the easiest way to do that is probably by these minimal wrappers.  That particular method was just one that was missing from the collection of wrappers that already existed for this purpose.

> Is this with some sort of future changes in mind?

Yes.  I would like to move to a context which we manage st that if you call our `random` module methods it not only pushes the external context to the side, but by default iterates our own internal context instead of _constantly resetting it_. I think this would also be the same path to get us out of legacy randomstate, and into the current numpy `Generator/BitGenerator` framework, conveniently enough... and it would allow having k sequences for k threads etc.... but it has a hang up...

It unfortunately would likely change our test results etc to get us there, so I was just proposing this as some prep steps that should be otherwise harmless atm. I think it would be better to tackle those other items separately from the more harmless consistency/module-name cleanup here.

A big picture idea would be that if our Random context was extended then:
```
with Random:
  npRandThing1
  npRandThing2
```

would by default would be equiv to 
```
with Random:
  npRandThing1
with Random
  npRandThing2
```

and also
```
aspireWrappedRandThing1
aspireWrappedRandThing2
```

You could of course reset/force things, but generally speaking I think the above (combined with not harming anybody else's state) would be a pretty solid default approach. It is more robust than what we do now (I think).

File: src/aspire/utils/coor_trans.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/coor_trans.py]: See the longer note below. I think by default our Random tools should effectively manage the situation in a way that those two invocations would be equivalent... and would prefer just use our wrappers to manage the contexts than developers coding the contexts.

IME it is actually important to iterate the RNG in practice.  What we do now (fine for UT I guess, maybe...) resets RNG virtually all the time.  My stance is that we only want to reset under very specific and explicit circumstances.

File: src/aspire/utils/coor_trans.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/utils/coor_trans.py]: yes, right, only in comment.  1ac4268b509c5064f1a47a51ed569f4bf32b04ee

File: src/aspire/utils/coor_trans.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/coor_trans.py]: I removed this comment for now, citing this discussion. I do think there is still a lot of work todo wrt Random, but I was mainly just trying to get a little bit of work done in that direction between other larger (or blocked) tasks.

**Comment by garrettwrong** [comment]
This has been rebased with the tox/black changes.

**Review by janden — APPROVED** [review]

**Pull request merged**